### PR TITLE
Story/14376 Handle SRS server warning for truncated keys

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
Handle SRS server warning for truncated keys

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14376

### CODE REVIEW

**Review by KathrinLuecke — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 1]
```diff
@@ -1676,6 +1684,25 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 			alert.preferredAction = confirmAction
 			
 			viewController.present(alert, animated: true)
+		case .increasedVolumeOfWarnings(let model):
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]:
```suggestion
		case .increasedWarningsVolume(let model):
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 2]
```diff
@@ -1036,7 +1036,11 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 					
 					self.model.shouldShowSymptomsOnsetScreen ? self.showSymptomsOnsetScreen() : self.submitExposure(showSubmissionSuccess: true, isLoading: isLoading)
 				case .srs:
-					self.model.shouldShowSymptomsOnsetScreen ? self.showSymptomsOnsetScreen() : self.submitSRSExposure(showSubmissionSuccess: true, isLoading: isLoading)
+					self.model.shouldShowSymptomsOnsetScreen ? self.showSymptomsOnsetScreen() : self.submitSRSExposure(
+						showSubmissionSuccess: true,
+						viewControllerToPresentingAlert: exposureSubmissionSymptomsViewController,
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:1041]:
```suggestion
						presentingViewController: exposureSubmissionSymptomsViewController,
```

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 3]
```diff
@@ -1049,6 +1049,12 @@ Bei ausgeschalteter Hintergrundaktualisierung müssen Sie die App täglich aufru
 
 "ExposureSubmission_SRSFlow_alert_confirmWarnOthers_ActionCancel" = "Abbrechen";
 
+"ExposureSubmission_SRSFlow_alert_increasedVolumeOfWarnings_Title" = "Hinweis";
+
+"ExposureSubmission_SRSFlow_alert_increasedVolumeOfWarnings_Message" = "Aufgrund des aktuell erhöhten Aufkommens von Warnungen über die App wurden nur Kontakte der letzen %@ Tage gewarnt.";
+
+"ExposureSubmission_SRSFlow_alert_increasedVolumeOfWarnings_ActionOkay" = "OK";
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:1056]: @KathrinLuecke is this fine for you? "OK"

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/KeySubmission/SRSKeySubmissionResource.swift [hunk 4]
```diff
@@ -7,6 +7,7 @@ import Foundation
 enum SRSKeySubmissionResourceError: LocalizedError, Equatable {
 	case invalidPayloadOrHeader
 	case invalidOtp
+	case tooManyRequestsKeyPerDay
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClient/Resources/KeySubmission/SRSKeySubmissionResource.swift]:
```suggestion
	case tooManyKeyRequestsPerDay
```

File: src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Resource/ReceiveResources/SRSSuccessReceiveResource.swift [hunk 5]
```diff
@@ -0,0 +1,16 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+struct SRSSuccessReceiveResource: ReceiveResource {
+	
+	typealias ReceiveModel = Int?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Resource/ReceiveResources/SRSSuccessReceiveResource.swift:9]: Missing mark

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 6]
```diff
@@ -375,6 +375,12 @@ enum AppStrings {
 		static let actionConfirm = NSLocalizedString("ExposureSubmission_SRSFlow_alert_confirmWarnOthers_ActionConfirm", comment: "")
 		static let actionCancel = NSLocalizedString("ExposureSubmission_SRSFlow_alert_confirmWarnOthers_ActionCancel", comment: "")
 	}
+	
+	enum SRSIncreasedVolumeOfWarningsAlert {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift]:
```suggestion
	enum SRSIncreasedWarningsVolumeAlert {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 7]
```diff
@@ -2039,6 +2087,13 @@ extension ExposureSubmissionCoordinator {
 		enum Consent {
 			case cancelWarnOthers(on: UIViewController)
 			case confirmWarnOthers(on: UIViewController)
+			case increasedVolumeOfWarnings(IncreasedVolumeOfWarningsModel)
+
+			struct IncreasedVolumeOfWarningsModel {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]:
```suggestion
			struct IncreasedWarningsVolumeModel {
```

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 8]
```diff
@@ -1869,16 +1896,37 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
...
+	) {
 		self.model.submitSRSExposure(
 			isLoading: isLoading,
-			onSuccess: { [weak self] in
+			onSuccess: { [weak self] cwaKeyTruncated in
 				self?.store.mostRecentKeySubmissionDate = Date()
 
-				if showSubmissionSuccess {
-					self?.showExposureSubmissionSuccessViewController()
+				let continueHandler = { [weak self] in
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:1909]: nextAction

Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]:
```suggestion
				let nextAction = { [weak self] in
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 9]
```diff
@@ -1676,6 +1684,25 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
...
+			
+			alert.addAction(UIAlertAction(
+				title: AppStrings.SRSIncreasedVolumeOfWarningsAlert.actionOkay,
+				style: .default,
+				handler: { _ in
+					model.okayHandler()
+				}
+			))
+			
+			model.viewControllerToPresentingAlert.present(alert, animated: true)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]:
```suggestion
			model.presentingViewController.present(alert, animated: true)
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 10]
```diff
@@ -1869,16 +1896,37 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
...
 		self.model.submitSRSExposure(
 			isLoading: isLoading,
-			onSuccess: { [weak self] in
+			onSuccess: { [weak self] cwaKeyTruncated in
 				self?.store.mostRecentKeySubmissionDate = Date()
 
-				if showSubmissionSuccess {
-					self?.showExposureSubmissionSuccessViewController()
+				let continueHandler = { [weak self] in
+					if showSubmissionSuccess {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:1910]: Cleaner to use shorthand property? 

showSubmissionSuccess ? self?.showExposureSubmissionSuccessViewController() : self?.dismiss()

Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:1910]: In cases of value assigning, yes. In this case, I guess a classical if-else is good readable. Would keep that.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 1 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: changed

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 2 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:1041]: changed

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 3 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings:1056]: approved form her

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/KeySubmission/SRSKeySubmissionResource.swift (hunk 4 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/KeySubmission/SRSKeySubmissionResource.swift]: changed

File: src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Resource/ReceiveResources/SRSSuccessReceiveResource.swift (hunk 5 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Resource/ReceiveResources/SRSSuccessReceiveResource.swift:9]: added

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift (hunk 6 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift]: changed

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 7 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: changed

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 8 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: changed

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 9 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: changed

**Review by puneetmahali1 — APPROVED** [review]

**Pull request merged**